pkg/notes/notes.go

```diff
@@ -399,32 +357,28 @@ func ReleaseNoteFromCommit(result *Result, relVer string, opts ...GitHubAPIOptio
 // ListCommits lists all commits starting from a given commit SHA and ending at
 // a given commit SHA.
 func (g *Gatherer) ListCommits(branch, start, end string) ([]*github.RepositoryCommit, error) {
-	c := configFromOpts(g.Opts...)
-
-	c.branch = branch
-
-	startCommit, _, err := g.Client.GetCommit(c.ctx, c.org, c.repo, start)
+	startCommit, _, err := g.Client.GetCommit(g.Context, g.Org, g.Repo, start)
 	if err != nil {
 		return nil, err
 	}
 
-	endCommit, _, err := g.Client.GetCommit(c.ctx, c.org, c.repo, end)
+	endCommit, _, err := g.Client.GetCommit(g.Context, g.Org, g.Repo, end)
 	if err != nil {
 		return nil, err
 	}
 
 	allCommits := &commitList{}
 
-	worker := func(cl *commitList, clo *github.CommitsListOptions) (*github.Response, error) {
-		c, resp, err := g.Client.ListCommits(c.ctx, c.org, c.repo, clo)
-		if err == nil {
-			cl.Add(c)
+	worker := func(clo *github.CommitsListOptions) ([]*github.RepositoryCommit, *github.Response, error) {
+		commits, resp, err := g.Client.ListCommits(g.Context, g.Org, g.Repo, clo)
+		if err != nil {
+			return nil, nil, err
 		}
-		return resp, err
+		return commits, resp, err
 	}
 
 	clo := github.CommitsListOptions{
-		SHA:   c.branch,
+		SHA:   branch,
 		Since: *startCommit.Committer.Date,
 		Until: *endCommit.Committer.Date,
```

Comment:
I assume this can never be `nil` so it's fine, right?

Reply:
I certainly could be `nil`. Those structs are built with expressing everything as pointers. Also it has been like that all along.
However, the structs also have methods to access that data, so I switched to using that.